toolset/utils/docker_helper.py

```diff
@@ -124,272 +129,238 @@ def build(benchmarker_config, test_names, build_log_dir=os.devnull):
+
+            extra_hosts = None
+            name = "tfb-server"
+
+            if self.benchmarker.config.network is None:
+                extra_hosts = {
+                    socket.gethostname(): str(self.benchmarker.config.server_host),
+                    'tfb-server': str(self.benchmarker.config.server_host),
+                    'tfb-database': str(self.benchmarker.config.database_host)
+                }
+                name = None
+
+            sysctl = {'net.core.somaxconn': 65535}
+
+            ulimit = [{
+                'name': 'nofile',
+                'hard': 200000,
+                'soft': 200000
+            }, {
+                'name': 'rtprio',
+                'hard': 99,
+                'soft': 99
+            }]
+
+            container = self.server.containers.run(
+                "techempower/tfb.test.%s" % test.name,
+                name=name,
+                network=self.benchmarker.config.network,
+                network_mode=self.benchmarker.config.network_mode,
+                stderr=True,
+                detach=True,
+                init=True,
+                extra_hosts=extra_hosts,
+                privileged=True,
+                ulimits=ulimit,
+                sysctls=sysctl,
+                remove=True,
+                log_config={'type': None})
+
+            watch_thread = Thread(
+                target=watch_container,
+                args=(
+                    container,
+                    "%s.dockerfile" % test.name,
+                ))
+            watch_thread.daemon = True
+            watch_thread.start()
 
-        def watch_container(docker_container, docker_file):
+        except Exception:
             with open(
-                    os.path.join(run_log_dir, "%s.log" % docker_file.replace(
-                        ".dockerfile", "").lower()), 'w') as run_log:
-                for line in docker_container.logs(stream=True):
-                    log(line, prefix=log_prefix, file=run_log)
-
-        extra_hosts = None
-        name = "tfb-server"
-
-        if benchmarker_config.network is None:
-            extra_hosts = {
-                socket.gethostname(): str(benchmarker_config.server_host),
-                'tfb-server': str(benchmarker_config.server_host),
-                'tfb-database': str(benchmarker_config.database_host)
-            }
-            name = None
+                    os.path.join(run_log_dir, "%s.log" % test.name.lower()),
+                    'w') as run_log:
+                tb = traceback.format_exc()
+                log("Running docker cointainer: %s.dockerfile failed" % test.name,
+                    prefix=log_prefix,
+                    file=run_log)
+                log(tb, prefix=log_prefix, file=run_log)
 
-        sysctl = {'net.core.somaxconn': 65535}
+        return container
+
+    @staticmethod
+    def kill(container):
+        try:
+            container.kill()
```

Comment:
Why is `container.kill` instead of `container.stop`?

Reply:
stop sends `SIGTERM` to the container and then after a grace period sends `SIGKILL`, this _always_ leads to a long wait on ctrl-c and errors because it always ends up sending `SIGKILL`. `container.kill` sends `SIGKILL` immediately. `ctrl-c` works instantly now.

Reply:
Okay, that's fine for `ctrl-c`, but the normal stop should still be `stop`, right?